### ISSUE TITLE
efibootmgr: No such bootvar BootFFFFFFF4

### ISSUE THREAD

**Issue by HSarham** [project-trident/trident-installer#46]
I get the following error when I try to install project trident on my laptop:
![5C666EEF-3B8F-4EB7-9139-2539B598DFA0](https://user-images.githubusercontent.com/39811387/55676533-770c0480-58d7-11e9-9d15-aa88e2f37992.jpeg)

**Comment by rkitover** [comment]
I just got this too, here is some more context from the log:

```
Running: mkdir -p /mnt/boot/efi/EFI/BOOT
Running: kldload efirt
kldload: can't load efirt: module already loaded or in kernel
Running: cp /mnt/boot/loader.efi /mnt/boot/efi/EFI/BOOT/BOOTX64-TRUEOS.EFI
Running: efibootmgr -a -c -l /mnt/boot/efi/EFI/BOOT/BOOTX64-TRUEOS.EFI -L TrueOS
efibootmgr: No such bootvar BootFFFFFFF4
: No such file or directory
EXITERROR: Error 1: efibootmgr -a -c -l /mnt/boot/efi/EFI/BOOT/BOOTX64-TRUEOS.EFI -L TrueOS
ZFS Unmount: doobie/var/tmp
```

is this an issue with the latest image perhaps?

**Comment by SnowCrestGuy** [comment]
This is caused by trying to activate a bootvar prior to that var being created. The "-a" is the error. Also could the UEFI partition directory structure be change from the default location to "/EFI/True0s/BOOTX64-TRUEOS.EFI? The /EFI/BOOT/xxxxxxx.efi structure is only required for removable media. Linux and Win10 already use their own directory. The update programs will need to be fixed to look for the correct location of the UEFI partition data. Doing the directory change will alleviate other program installs from trashing the UEFI data.

**Comment by HSarham** [comment]
I think the problem is that no BootVarNum is specified after the “-a” option. According to the [manual page of efibootmgr](https://www.freebsd.org/cgi/man.cgi?query=efibootmgr&apropos=0&sektion=0&manpath=FreeBSD+13-current&arch=default&format=html) it is required:
“-a	--activate BootVarNum	   Activate the	given BootVarNum.”

**Comment by rkitover** [comment]
I tried installing in legacy mode with CSM, but then the scfb xorg driver does not work and there is no way to start the installer. The vesa driver does not work error, says something about an error opening vbios. Why is the intel driver not on the installer image?

**Comment by rkitover** [comment]
So as I found out from the telegram group, this `efibootmgr` command is the last in the install process, and you can complete it by running it without the `-a` option, like so:

- click "GO TO TERMINAL" or reboot to the installer and click "GO TO TERMINAL" if you have already installed and have the install disk
- mount the ESP created by the installer, e.g.:
```bash
mount_msdosfs /dev/nvd1p1 /mnt
```
replace `nvd1p1` with the device of the created ESP (EFI partition)
- go to the location of the `.efi` file
```bash
cd /mnt/EFI/BOOT
```
- run the `efibootmgr` command without the `-a` option:
```bash
efibootmgr -c -l BOOTX64-TRUEOS.EFI -L "Project Trident"
```
- reboot to your bios, go to "HARD DRIVE BBS PRIORITIES", and choose "Project Trident" as the first option
- boot to your shiny new project trident install

Hope this helps anyone who comes across this.

**Comment by HSarham** [comment]
Very nice, thanks for the solution, however it would be better if the issue gets fixed in the next update.

**Comment by kmoore134** [comment]
Updated pc-sysinstall upstream, the -a has been removed.

**Comment by rkitover** [comment]
Thank you!

**Comment by beanpole135** [comment]
I have updated the trident installer to pull in the newer version of pc-sysinstall from ports and use that (instead of the one in the base system).
Waiting to see the updated port become available before I start the next build, so this can be included in the 18.12-U9 update.

Marking issue as fixed.
